File: capacity-club-api/src/domain-modules/home-training/model/payload/home-training-create.payload.ts
import { Payload } from '@domain-modules-shared';
import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsOptional } from 'class-validator';
import { MemberHomeTraining } from 'domain-modules/member-home-training/model/entity/member_home_training.entity';
import { Workout } from 'domain-modules/workout/model/entity/workout.entity';

/**
 * Class used to create home training.
 * This class defines the payload used when creating new home training.
 */
export class HomeTrainingCreatePayload implements Payload {
  @ApiProperty()
  @IsNotEmpty()
  title: string;
  @ApiProperty()
  @IsOptional()
  description: string;
  @ApiProperty()
  @IsNotEmpty()
  nb_week: number;
  @ApiProperty()
  @IsNotEmpty()
  nb_training_by_week: number;
  @ApiProperty()
  @IsNotEmpty()
  price: number;
  @ApiProperty({ type: () => Workout, isArray: true })
  @IsNotEmpty()
  workouts: Workout[];
  @ApiProperty({ type: () => MemberHomeTraining, isArray: true })
  @IsNotEmpty()
  member_home_trainings: MemberHomeTraining[];
}

File: capacity_club_mobile_app/lib/domain-features/data/home-training/model/payload/home_training_create_payload.dart
import 'package:capacity_club_mobile_app/core/model/abstract/payload.dart';
import 'package:capacity_club_mobile_app/domain-features/data/workout/model/workout_model.dart';
import 'package:equatable/equatable.dart';

class HomeTrainingCreatePayload extends Equatable implements Payload {
  final String title;
  final int nb_week;
  final int nb_training_by_week;
  final double price;
  final List<WorkoutModel> workouts;

  const HomeTrainingCreatePayload({
    required this.title,
    required this.nb_week,
    required this.nb_training_by_week,
    required this.price,
    required this.workouts,
  });

  Map<String, dynamic> toJson() => {
        'title': title,
        'nb_week': nb_week,
        'nb_training_by_week': nb_training_by_week,
        'price': price,
        'workouts': workouts.map((workout) => workout.toJson()).toList(),
      };

  @override
  List<Object?> get props => [
        title,
        nb_week,
        nb_training_by_week,
        price,
        workouts,
      ];
}
